react-accordion-component/my-app/src/Accordion.tsx

```diff
@@ -0,0 +1,57 @@
+import { MouseEventHandler, useState } from 'react';
+import './Accordion.css';
+
+export type Topic = {
+  title: string;
+  content: string;
+};
+
+type ParentProps = {
+  topics: Topic[];
+};
+
+type ChildProps = {
+  topic: Topic;
+  onToggle: MouseEventHandler;
+  currentItem: boolean;
+};
+
+export default function Accordion({ topics }: ParentProps) {
+  const [openItem, setOpenItem] = useState<null | number>(null);
```

Comment:
If you use `undefined` instead of `null`, the default state value is `undefined`, so you can just declare this as:
`const [openItem, setOpenItem] = useState<number>();`
and the type definitions will result in `openItem` having the type `number | undefined`.